invoices_service.go

```diff
@@ -140,6 +140,25 @@ func (s *invoicesImpl) Create(accountCode string, invoice Invoice) (*Response, *
 	return resp, &dst, err
 }
 
+// Collect force retries the card on file for the invoice.
+// Allows to collect a past-due or pending invoice. This API is rate limited
+// and only one collection attempt per account is allowed within an hour.
+// Will return status code 400 if rate limit hit or if invoice not in the
+// correct state.
+// https://dev.recurly.com/v2.5/docs/collect-an-invoice
+func (s *invoicesImpl) Collect(invoiceNumber int) (*Response, *Invoice, error) {
+	action := fmt.Sprintf("invoices/%d/collect", invoiceNumber)
+	req, err := s.client.newRequest("PUT", action, nil, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var dst Invoice
+	resp, err := s.client.do(req, &dst)
```

Comment:
Could you return a `nil` invoice and add a corresponding test when the invoice isn't retrieved? See this PR for reference: https://github.com/blacklightcms/recurly/pull/19